lib/user_input_validator.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+# UserInputValidator
+class UserInputValidator
+  def move_valid_integer?(input_string)
+    # Requires an integer or string
+    if (!input_string.is_a? Integer) && (!input_string.is_a? String)
+      return false
+    end
+
+    # Strings will be converted to integer using Integer
+    # Valid strings: '1', '  1 ', ' 2 \n', ' 2 \r\n'
+    # Invalid strings: '', 'aa', 'a1', '1a', '1 2', '1.1'
+    begin
+      input_string = Integer input_string
+    rescue ArgumentError
+      return false # Failed to parse String into Integer
+    end
+
+    return false if input_string < 1 || input_string > 9
+
```

Comment:
And it is a good practice to try to avoid using "magi" strings and numbers. I mean it is not obvious here what are `1` and `9`